### ISSUE TITLE
android.location.Geocoder getFromLocation(double, double, int) was deprecated in API level 33

### ISSUE THREAD

**Issue by hongeSunCoder** [Baseflow/flutter-geocoding#109]

**Comment by mvanbeusekom** [comment]
Providing some additional context:

The current version of the geocoding_android package makes use of the `List<Address> getFromLocation(double, double, int)` method on the following lines:

- [Geocoding.java line 36](https://github.com/Baseflow/flutter-geocoding/blob/39ce642b1e8a70728e16f9d74c3ce92480b2b304/geocoding_android/android/src/main/java/com/baseflow/geocoding/Geocoding.java#L36)
- [Geocoding.java line 54](https://github.com/Baseflow/flutter-geocoding/blob/39ce642b1e8a70728e16f9d74c3ce92480b2b304/geocoding_android/android/src/main/java/com/baseflow/geocoding/Geocoding.java#L54)

As per Androids [documentation](https://developer.android.com/reference/android/location/Geocoder#getFromLocation(double,%20double,%20int)) this method has been deprecated in Android SDK 33, and should be replaced by `void getFromLocation(double, double, int, Geocoder.GeocodeListener)`.

Basically this means the method moved from a synchronous to an asynchronous signature, which requires a bigger update. We will address this in a larger overhaul, in the meantime it is save to ignore this warning as the deprecated method is still available.

**Comment by hansajith1998** [comment]
![image](https://github.com/Baseflow/flutter-geocoding/assets/64520856/016c160a-d74a-4a0d-b769-371ff5098687)

**Comment by mvanbeusekom** [comment]
This problem should be resolved with version 5.0.0 of the geocoding plugin.
